File: app/src/main/java/com/krystianwsul/checkme/domainmodel/local/LocalWeeklyScheduleBridge.java
package com.krystianwsul.checkme.domainmodel.local;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.krystianwsul.checkme.domainmodel.WeeklyScheduleBridge;
import com.krystianwsul.checkme.persistencemodel.ScheduleRecord;
import com.krystianwsul.checkme.persistencemodel.WeeklyScheduleRecord;
import com.krystianwsul.checkme.utils.CustomTimeKey;

class LocalWeeklyScheduleBridge extends LocalScheduleBridge implements WeeklyScheduleBridge {
    @NonNull
    private final WeeklyScheduleRecord mWeeklyScheduleRecord;

    LocalWeeklyScheduleBridge(@NonNull ScheduleRecord scheduleRecord, @NonNull WeeklyScheduleRecord weeklyScheduleRecord) {
        super(scheduleRecord);

        mWeeklyScheduleRecord = weeklyScheduleRecord;
    }

    @Override
    public int getDayOfWeek() {
        return mWeeklyScheduleRecord.getDayOfWeek();
    }

    @Nullable
    @Override
    public CustomTimeKey getCustomTimeKey() {
        if (mWeeklyScheduleRecord.getCustomTimeId() != null)
            return new CustomTimeKey(mWeeklyScheduleRecord.getCustomTimeId());
        else
            return null;
    }

    @Nullable
    @Override
    public Integer getHour() {
        return mWeeklyScheduleRecord.getHour();
    }

    @Nullable
    @Override
    public Integer getMinute() {
        return mWeeklyScheduleRecord.getMinute();
    }

    @Override
    public void delete() {
        mScheduleRecord.delete();
        mWeeklyScheduleRecord.delete();
    }
}

File: app/src/main/java/com/krystianwsul/checkme/domainmodel/local/LocalWeeklyScheduleBridge.kt
package com.krystianwsul.checkme.domainmodel.local

import com.krystianwsul.checkme.domainmodel.WeeklyScheduleBridge
import com.krystianwsul.checkme.persistencemodel.ScheduleRecord
import com.krystianwsul.checkme.persistencemodel.WeeklyScheduleRecord
import com.krystianwsul.checkme.utils.CustomTimeKey

internal class LocalWeeklyScheduleBridge(scheduleRecord: ScheduleRecord, private val mWeeklyScheduleRecord: WeeklyScheduleRecord) : LocalScheduleBridge(scheduleRecord), WeeklyScheduleBridge {

    override val daysOfWeek get() = setOf(mWeeklyScheduleRecord.dayOfWeek)

    override val customTimeKey get() = mWeeklyScheduleRecord.customTimeId?.let { CustomTimeKey(it) }

    override val hour get() = mWeeklyScheduleRecord.hour

    override val minute get() = mWeeklyScheduleRecord.minute

    override fun delete() {
        scheduleRecord.delete()
        mWeeklyScheduleRecord.delete()
    }
}
